tests/measurements/mpi_counter_test.cpp

```diff
@@ -0,0 +1,189 @@
+// This file is part of KaMPIng.
+//
+// Copyright 2024 The KaMPIng Authors
+//
+// KaMPIng is free software : you can redistribute it and/or modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later
+// version. KaMPIng is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the
+// implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+// for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License along with KaMPIng.  If not, see
+// <https://www.gnu.org/licenses/>.
+
+#include "gmock/gmock.h"
+#include <algorithm>
+#include <cstddef>
+#include <stack>
+#include <tuple>
+#include <unordered_map>
+
+#include <gtest/gtest.h>
+#include <mpi.h>
+
+#include "kamping/measurements/counter.hpp"
+#include "measurement_test_helpers.hpp"
+
+using namespace ::kamping;
+using namespace ::testing;
+using namespace ::measurements;
+
+TEST(CounterTest, basics) {
+    auto const& comm = comm_world();
+    Counter     counter;
+    using DataType = decltype(counter)::DataType;
+    counter.add("measurement", 42);
+    auto                        aggregated_counter_tree = counter.aggregate();
```

Comment:
I am not sure if return the max is the default behavior I would expect for a scalar when aggregating globally. I think I would expect the sum to be the default?